toke.c

```diff
@@ -6855,6 +6842,21 @@ Perl_yylex(pTHX)
         return yyl_sigvar(aTHX_ s);
     }
 
+    return yyl_try(aTHX_ 0, s, 0, 0, NULL, NULL, 0, 0, saw_infix_sigil);
+}
+
+static int
+yyl_try(pTHX_ char initial_state, char *s, STRLEN len,
+        I32 orig_keyword, GV *gv, GV **gvp,
+        U8 formbrack, U32 fake_eof, const bool saw_infix_sigil)
+{
+    char *d;
+    bool bof = FALSE;
+
+    switch (initial_state) {
+    case '}': goto rightbracket;
+    }
```

Comment:
weird switch, but ok

Reply:
I was thinking there would end up being other situations where a call would need to jump further down, especially once the trickier bits of what's now `yyl_try()` are tackled. But I'm happy to rework this if you think that would be an improvement?

Reply:
this is fine as it :-) I was just pointing it, right now it s more a boolean context
but I can also see further usage in the future